packages/api/src/routes/nfts-upload.js

```diff
@@ -105,9 +111,48 @@ export async function nftUpload(event, ctx) {
     })
   }
 
+  if (ctx.w3up) {
+    if (!w3upFeatureSwitchEnabled(ctx, { user })) {
+      console.warn(`skipping w3up upload. Feature switch does not allow it.`)
+    } else {
+      try {
+        await ctx.w3up.uploadCAR(carBlobForW3up)
+      } catch (error) {
+        // explicitly log so we can debug w/ cause
+        console.warn('error with w3up.uploadCAR', {
+          error,
+          cause: /** @type any */ (error)?.cause,
+        })
+        throw error
+      }
+    }
+  } else {
+    console.warn(
+      'nftUpload route skipping w3up storage due to missing ctx.w3up'
+    )
+  }
```

Comment:
This needs to be pushed into `uploadCar` and the upload to w3up should happen _instead_ of upload to s3/r2. There is more than one route that calls this. e.g. https://github.com/nftstorage/nft.storage/blob/main/packages/api/src/routes/nfts-store.js

Reply:
ah yea - I think this was intentional and intended to limit the potential scope of changes - is there any downside to pushing the data through both upload paths for now, aside from some extra computation and associated $ costs? If the only problem is that doing both uploads will result in extra infra costs for the next couple months, and we can limit potentially disruptive changes then I'm in favor of just having this code do the upload both ways and then potentially removing the "classic" upload path once we're confident things are working... what do you think?

Reply:
I'm probbaly more pro replacing it. For one it makes it easier to test if everything is working or not - no false positives since it's not going through the old pipeline as well. It also saves money. I think it mostly boils down to a conditional over https://github.com/nftstorage/nft.storage/blob/06712ad02d8df58b654b77a3a4db0eb5fa213fb7/packages/api/src/routes/nfts-upload.js#L152-L155 ...but then linkdex API will not be able to determine completeness since it won't be put in S3. I reckon we could quite easily deploy an equivalent service that works on cloudflare and looks at R2.

Reply:
ok! I started to look into this but I think it's a decent chunk of work and I wouldn't be able to finish before the end of the day, so I'll leave this one for you tomorrow!